lib/lotus/mailer/rendering.rb

```diff
@@ -4,17 +4,35 @@ module Mailer
     #
     # @since 0.1.0
     module Rendering
-
-      # Render a single template with the specified format.
-      #
-      # @return [String] the output of the rendering process.
-      #
-      # @since 0.1.0
-      def render(format)
-        templates
-        @templates[format].render
+      def self.extended(base)
+        base.class_eval do
+          include InstanceMethods
+        end
       end
 
+      module InstanceMethods
+        # Initialize a mailer
+        #
+        # @param locals [Hash] a set of objects available during the rendering process.
+        #
+        # @since 0.1.0
+        def initialize(locals = {})
+          @locals   = locals
+          @scope    = self
+        end
+
+        # Render a single template with the specified format.
+        #
+        # @param format [symbol] template's format
+        #
+        # @return [String] the output of the rendering process.
+        #
+        # @since 0.1.0
+        def render(format)
+          self.class.templates
```

Comment:
Because it may not have loaded the templates before (searched for the ones with the same name), in which case when we tried to access them it would give an error.
This should be done within the load method but we would have problems with the tests if that was the case because of the freezing ;)